### PR TITLE
Enhanced RDoc for IO

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Adds  sections to class RDoc:
- Lines
  - Line Separator
  - Line Limit
  - Line Number
 
Revises example text file `t.txt` to:
- Include paragraphs (separated by double line separator).
- Avoid being too long.

Revises examples that use the changed example file.

There are several other methods that will point to the added sections.

### CODE REVIEW

File: io.c [hunk 1]
```diff
@@ -3975,70 +3974,60 @@ rb_io_gets_internal(VALUE io)
...
  *    f = File.open('t.txt')
  *    # Chomp the lines.
- *    f.gets(chomp: true) # => "This is line one."
- *    f.gets(chomp: true) # => "This is the second line."
- *    f.gets(chomp: true) # => "This is the third line."
+ *    f.gets(chomp: true) # => "First line"
+ *    f.gets(chomp: true) # => "Second line"
+ *    f.gets(chomp: true)# => ""
+ *    f.gets(chomp: true)  # => "Fourth line"
+ *    f.gets(chomp: true)      # => "Fifth line"
```
Comment (marivaldo) [inline io.c:4030]:
```suggestion
 *    f.gets(chomp: true) # => ""
 *    f.gets(chomp: true) # => "Fourth line"
 *    f.gets(chomp: true) # => "Fifth line"
```

File: io.c [hunk 2]
```diff
@@ -3461,19 +3460,19 @@ io_write_nonblock(rb_execution_context_t *ec, VALUE io, VALUE str, VALUE ex)
  *  When argument +out_string+ is given,
  *  the returned value is +out_string+, whose content is replaced:
  *
- *    f = File.new('t.txt')
+ *    f = File.ne   w('t.txt')
```
Comment (marivaldo) [inline io.c:3463]:
```suggestion
 *    f = File.new('t.txt')
```

File: io.c [hunk 3]
```diff
@@ -3975,70 +3974,60 @@ rb_io_gets_internal(VALUE io)
...
- *    f.gets(' is') # => "This is"
- *    f.gets(' is') # => " line one.\nThis is"
- *    f.gets(' is') # => " the second line.\nThis is"
- *    f.gets(' is') # => " the third line.\n"
- *    f.gets(' is') # => nil
+ *    f = File.new('t.txt')
+ *    f.gets('l')    # => "First l"
+ *    f.gets('li')   # => "ine\nSecond li"
+ *    f.gets('lin')  # => "ne\n\nFourth lin"
+ *    f.gets         # => "e\n"
```
Comment (marivaldo) [inline io.c:4001]:
```suggestion
 *    f.gets('l')   # => "First l"
 *    f.gets('li')  # => "ine\nSecond li"
 *    f.gets('lin') # => "ne\n\nFourth lin"
 *    f.gets        # => "e\n"
```

File: io.c [hunk 4]
```diff
@@ -13941,7 +13932,92 @@ set_LAST_READ_LINE(VALUE val, ID _x, VALUE *_y)
...
+ *    f.gets # => "Fourth line\n"
+ *    f.gets # => "Fifth line\n"
+ *
+ *  You can specify a different line separator:
+ *
+ *    f = File.new('t.txt')
+ *    f.gets('l')    # => "First l"
+ *    f.gets('li')   # => "ine\nSecond li"
+ *    f.gets('lin')  # => "ne\n\nFourth lin"
+ *    f.gets         # => "e\n"
```
Comment (marivaldo) [inline io.c:13975]:
```suggestion
 *    f.gets('l')   # => "First l"
 *    f.gets('li')  # => "ine\nSecond li"
 *    f.gets('lin') # => "ne\n\nFourth lin"
 *    f.gets        # => "e\n"
```

File: io.c [hunk 5]
```diff
@@ -13941,7 +13932,92 @@ set_LAST_READ_LINE(VALUE val, ID _x, VALUE *_y)
...
+ *
+ *  There are two special line separators:
+ *
+ *  - +nil+:  The entire stream is read into a single string:
```
Comment (marivaldo) [inline io.c:13979]:
```suggestion
 *  - +nil+: The entire stream is read into a single string:
```

File: io.c (hunk 1 above)
Comment (BurdetteLamar) [reply io.c]: Fixed

File: io.c (hunk 2 above)
Comment (BurdetteLamar) [reply io.c]: Fixed.

File: io.c (hunk 3 above)
Comment (BurdetteLamar) [reply io.c]: Fixed.

File: io.c (hunk 4 above)
Comment (BurdetteLamar) [reply io.c]: Fixed.

File: io.c (hunk 5 above)
Comment (BurdetteLamar) [reply io.c]: Fixed.

File: io.c [hunk 6]
```diff
@@ -13941,7 +13932,92 @@ set_LAST_READ_LINE(VALUE val, ID _x, VALUE *_y)
...
+ *    File.open('t.rus') {|f| f.gets(2).size } # => 1
+ *    File.open('t.rus') {|f| f.gets(3).size } # => 2
+ *    File.open('t.rus') {|f| f.gets(4).size } # => 2
+ *
+ *  With arguments +sep+ and +limit+,
+ *  combines the two behaviors:
+ *
+ *  - Returns the next line as determined by line separator +sep+,
+ *    or +nil+ if none.
+ *  - But returns no more than <tt>limit+1</tt> bytes.
```
Comment (jeremyevans) [inline io.c:14018]: This doesn't appear to be correct for multibyte IO:

```ruby
File.binwrite("t.t", "\u{10000}".b) # => 4
File.open('t.t', "r:UTF-8"){|f| f.gets("\n", 1).bytesize} #=> 4
```

The actual value is probably limit + maximum bytesize of single character in encoding - 1 (limit+3 for UTF-8, I'm not sure about other encodings).  Not sure if there is a succinct way of stating that.

**Review by jeremyevans — CHANGES_REQUESTED** [review]
Looks good, only one minor change requested.

File: io.c (hunk 6 above)
Comment (BurdetteLamar) [reply io.c]: Part of limit discussion recast.

File: io.c [hunk 7]
```diff
@@ -13941,7 +13932,92 @@ set_LAST_READ_LINE(VALUE val, ID _x, VALUE *_y)
...
- *  == Line Number
+ *  == Lines
+ *
+ *  Some reader methods in \IO are line-oriented;
+ *  such a method reads one or more lines,
+ *  which are separated by an implicit or explicit line separator.
+ *
+ *  These methods include:
+ *
+ *  - IO::foreach.
```
[the remark this answers is not in the document]
Comment (BurdetteLamar) [reply io.c:13943]: I like the periods.

**Review by jeremyevans — APPROVED** [review]
New wording looks good.  Thanks!

**Pull request merged**